server/protocols/jmap-rfc-8621/src/main/scala/org/apache/james/jmap/change/MailboxChangeListener.scala

```diff
@@ -94,10 +98,11 @@ case class MailboxChangeListener @Inject() (mailboxChangeRepository: MailboxChan
   }
 
   private def saveChangeEvent(jmapChange: JmapChange): Publisher[Void] =
-    jmapChange match {
+    SMono(jmapChange match {
       case mailboxChange: MailboxChange => mailboxChangeRepository.save(mailboxChange)
       case emailChange: EmailChange => emailChangeRepository.save(emailChange)
-    }
+    }).`then`(SMono(eventBus.dispatch(toStateChangeEvent(jmapChange), Set[RegistrationKey]().asJava)))
+
```

Comment:
no unit test for this part?

Reply:
IMO integration tests are enough. I suspect such unit tests to be hard to write, to infer too much about underlying data.